### ISSUE TITLE
Wrong base directory

### ISSUE THREAD

**Issue by rovo89** [rovo89/XposedInstaller#395]
Xposed stores files in `/data/data/de.robv.android.xposed.installer` (before Android 7.0 / Nougat) or  `/data/user_de/0/de.robv.android.xposed.installer` (since Android 7.0 Nougat). If this message is shown, the data directory provided by the system differs from this fixed location.

Possible reasons:

- Xposed Installer is installed on external storage.
**Solution:** Move to internal storage in Settings.
- You're not the primary user of the device. 
**Solution:** Switch to primary user.

The actual and expected paths are shown in the log file. Please post them here if neither of the solutions above worked for you.

**Comment by yaconsult** [comment]
I updated within xposed installer to version 88 and have this "wrong base directory" warning message on my Blu R1 HD running android 6.0.  The error message did not exist on the previous version of the xposed installer.  The xposed installer is not installed on external storage - no apps are on external storage.  I am the primary/only user of the device.  The log file is empty.

**Comment by rovo89** [comment]
I'm sorry to hear that. The check was mainly intended for Nougat, but I also added it to older Android versions for completeness. You should be able to install Xposed nevertheless, and after a reboot, the log should be filled. Would be great if you could post it then. I'll add the actual path to the log message for the next release, I missed the fact that the log isn't available of Xposed isn't installed yet.

**Comment by willstanelson** [comment]
On my tecno w3 version 6, that I just flashed in mm mod v4, I'm having this same error message and my WhatsApp can't work anymore it's stoping, don't really know if that's truly the cause but I know it's one of it. All the solutions above are not working... Please I need help. I think an update would be better. Thanks

**Comment by Massi-X** [comment]
Hi rovo, same issue on an always working marshmallow stock rom. It expects data/data but find data/user/0. The device is unencrypted, maybe this is because it fallback to data/user/0 instead of data/data? Only an idea.

**Comment by yaconsult** [comment]
Sorry that I wasn't more clear.  Xposed is already installed when I get this error message.  It happened after updating from the previous version to version 88.  The log file is empty, even after hitting the reload button.  Do I need to enable logging somewhere?

Again, Xposed is working properly.  The only issue from my perspective is the red error message on the Xposed screen complaining about the base directory.  I assume that the problem is in the check for the error condition since xposed continues to work properly after the update.

**Comment by leyan22** [comment]
warning message on the screen,but xposed is already worked. and xposed log is empty .

**Comment by aiolos01** [comment]
The new version displayed this warning. Xposed is working fine though. These are the directories from the log.
E/XposedInstaller( 9295): Base directory: /data/user/0/de.robv.android.xposed.installer
E/XposedInstaller( 9295): Expected: /data/data/de.robv.android.xposed.installer/

**Comment by XandrosDarkstorm** [comment]
Same thing happend to me today after updating from 82 to 88.2 and updating Xposed installer. Logs are empty. Android 6.0, Lenovo TB3-850M.

**Comment by rovo89** [comment]
Sorry, messed it up for older releases. [Xposed Installer 3.1.4](https://forum.xda-developers.com/showthread.php?p=74348202) fixes it.

**Comment by BlueMax** [comment]
Not fixed here. Now says 'Xposed not installed'. Still working properly though.
Nothing relevant in logs.

**Comment by oyxfishere** [comment]
3.1.4 fixed the issue for me, thanks a lot @rovo89

xposed-v88.2-sdk23-arm64.zip
redmi 4 prime(markw)
miui 8.5 chinese official stable
android 6.0.1

**Comment by BlueMax** [comment]
Anyone know where to download xposed installer version before 3.1.3? I've flushed userdata and now it not only says 'xposed not installed' but all modules are disabled as well. I'm locked out of xposed completely now.

This is the log after userdata has been flushed:
```
10-31 21:44:03.983 I/Xposed  (  578): -----------------
10-31 21:44:03.983 I/Xposed  (  578): Starting Xposed version 88.2 (Systemless by topjohnwu), compiled for SDK 23
10-31 21:44:03.983 I/Xposed  (  578): Device: GT-I9506 (samsung), Android version 6.0.1 (SDK 23)
10-31 21:44:03.983 I/Xposed  (  578): ROM: ks01ltexx-userdebug 6.0.1 MOI10E
10-31 21:44:03.983 I/Xposed  (  578): Build fingerprint: samsung/ks01ltexx/ks01lte:5.0.1/LRX22C/I9506XXUDOJ2:user/release-keys
10-31 21:44:03.983 I/Xposed  (  578): Platform: armeabi-v7a, 32-bit binary, system server: yes
10-31 21:44:03.983 I/Xposed  (  578): SELinux enabled: yes, enforcing: no
10-31 21:44:07.345 I/Xposed  (  578): -----------------
10-31 21:44:07.346 I/Xposed  (  578): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
10-31 21:44:07.864 I/Xposed  (  578): Detected ART runtime
10-31 21:44:07.883 I/Xposed  (  578): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
10-31 21:44:08.232 E/Xposed  (  578): Cannot load any modules because /data/data/de.robv.android.xposed.installer/conf/modules.list was not found
```

**Comment by brbsix** [comment]
@BlueMax

### rovo89's "official" installer
https://www.apkmirror.com/apk/rovo89/xposed-installer/

### DVDandroid's fork
https://forum.xda-developers.com/devdb/project/?id=13191#downloads

**Comment by yaconsult** [comment]
All fixed with the latest update - thank you, guys!!!

**Comment by zpp0196** [comment]
`cannot write /data/user_de/0/de.robv.android.xposed.installer.conf/modules/listjava.io/FileNotFoundException:/data/user_de/0/de.robv.android.xposed.installer/conf/modules.list(No such file or directory)`

* Xposed version:88.2
* Xposed Installer version:3.1.4
* Device:m3 note(Meizu),Android version 7.0(SDK 24)
* ROM: Flyme 6.10.30 beta/Flyme 6.7.11.28 beta/Flyme6.7.11.30 daily
* Platform:arm64-v8a,64-bit binary

What causes the bug, is it related to the Flyme system? Or Flyme's Android 7.0 system?

**Comment by h93910** [comment]
Can i use it without primary user?how to do

**Comment by rovo89** [comment]
> Can use it without primary user?how to do

No, only the primary user can manage Xposed. Apart from difficulties to manage per-user modules, it would also be a security risk, as modules are already loaded before the system is even starting up. Users aren't known at that time, therefore modules apply to all users.

**Comment by h93910** [comment]
ty , my friend
i use Android 8.1 is working , ty

**Comment by h93910** [comment]
okey , i get it.
i use the Android 8.1 is working
thanks for reply , my friend

**Comment by milzly** [comment]
It can not install xposed framework on Alcatel 4034n Xposed displays itself message Cp: can not create '/ system / xposed.prop': Read-only filesytem Error 1
